### DISCUSSION TITLE
Add username field to user account (after LDAP implementation, sAMAccountname) instead of using 'name'-Field

### DISCUSSION THREAD

**Discussion by TheMasterFX** [open-webui/open-webui#7298] [Issues]
# Feature Request
After the integration of LDAP authentication (thanks for that!), a new field ‘username’ should be implemented, since the LDAP user name usually does not correspond to the readable user name and also does not contain any spaces. e.g. Normally the uid/sAMAccountname is e.g. like lastname_firstname.
Now we have 
```
0|id|VARCHAR(255)|1||0
1|name|VARCHAR(255)|1||0
2|email|VARCHAR(255)|1||0
3|role|VARCHAR(255)|1||0
4|profile_image_url|TEXT|1||0
5|api_key|VARCHAR(255)|0||0
6|created_at|INTEGER|1||0
7|updated_at|INTEGER|1||0
8|last_active_at|INTEGER|1||0
9|settings|TEXT|0||0
10|info|TEXT|0||0
11|oauth_sub|TEXT|0||0
```
Maybe the uid should be a new field instead of the "name" which is the Full Name with email-login/registration. The Full name ('name') field is in Active Directory most times something like the givenName (sometimes  givenName + sn)

**Comment by wixyskywriter** [comment]
would it be possible to share the configuration details for LDAP integration with active directory?

**Comment by TheMasterFX** [comment]
![image](https://github.com/user-attachments/assets/16944da3-127d-48a2-b3bd-7f3393e915c2)
The Problem is, that the uid (or sAMAccountname) is not the real name in AD. So this is why it would be nice to have a Full name in the User Database (most times DisplayName)

**Reply by shihao138096** [reply]
you can try https://github.com/open-webui/open-webui/issues/11500

**Comment by TheMasterFX** [comment]
No interests on that? It is just an update of the the DB for a real Name (or first + last Name) in the database.
